Garage 2.0/Validation/SocialSecurityNumberAttribute.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace Garage_2._0.Validation
+{
+    public class SocialSecurityNumberAttribute : ValidationAttribute
+    {
+        const string errorMessage = "Enter in format: YYYYMMDD-xxxx";
+
+        protected override ValidationResult IsValid(object? value, ValidationContext validationContext)
+        {
+            if (value is string input)
+            {
+                var divided = input.Split('-');
+                //Checks that the format is right (only two parts after dividing with -)
+                if (divided.Length != 2)
+                    return new ValidationResult(errorMessage);
+
+                var dateOfBirth = divided[0];
+
+                //Checks that the length of the birth-date is correct
+                if (dateOfBirth.Length != 8)
+                    return new ValidationResult(errorMessage);
+
+                //Checks that it is only numbers
+                if (!int.TryParse(dateOfBirth, out _))
```

Comment:
I thought "Surely there must be a method for checking if a string is numeric" and was disapointed.

Reply:
Yup... You could think so